### PR TITLE
Adding pattern-4 and mysql dockerfile

### PR DESCRIPTION
**Pull request by pubudu538**
- Pattern-4 dockerfiles and artifacts
- Mysql dockerfile

### CODE REVIEW

File: pattern-3/artifacts/deploy-openshift.sh [hunk 1]
```diff
@@ -56,16 +56,16 @@ oc create -f apim-publisher/wso2apim-publisher-deployment.yaml
...
+#echo 'deploying apim key manager...'
+#oc create -f apim-km/wso2apim-km-deployment.yaml
+#
+#sleep 1m
+#echo 'deploying apim manager-worker ...'
+#oc create -f apim-gateway/wso2apim-manager-worker-deployment.yaml
+#
+#sleep 1m
+#echo 'deploying apim worker ...'
+#oc create -f apim-gateway/wso2apim-worker-deployment.yaml
```
Comment (isurulucky) [inline pattern-3/artifacts/deploy-openshift.sh]: Is this section commented out by a mistake?

File: pattern-3/dockerfiles/build.sh [hunk 2]
```diff
@@ -12,12 +12,12 @@ tm_1_dir=$(cd "${this_dir}/apim-tm-1"; pwd)
...
 docker build -t wso2/wso2am-publisher-pattern-3:2.1.0 $publisher_dir --squash
-docker build -t wso2/wso2am-store-pattern-3:2.1.0 $store_dir --squash
-docker build -t wso2/wso2am-tm-1-pattern-3:2.1.0 $tm_1_dir --squash
-docker build -t wso2/wso2am-tm-2-pattern-3:2.1.0 $tm_2_dir --squash
-docker build -t wso2/wso2am-km-pattern-3:2.1.0 $km_dir --squash
\ No newline at end of file
+#docker build -t wso2/wso2am-store-pattern-3:2.1.0 $store_dir --squash
+#docker build -t wso2/wso2am-tm-1-pattern-3:2.1.0 $tm_1_dir --squash
+#docker build -t wso2/wso2am-tm-2-pattern-3:2.1.0 $tm_2_dir --squash
+#docker build -t wso2/wso2am-km-pattern-3:2.1.0 $km_dir --squash
```
Comment (isurulucky) [inline pattern-3/dockerfiles/build.sh]: Keep squash by default.

File: pattern-3/artifacts/deploy-openshift.sh (hunk 1 above)
Comment (pubudu538) [inline pattern-3/artifacts/deploy-openshift.sh]: Fixed with https://github.com/wso2/kubernetes-apim/pull/19/commits/6627082c93cd02a92d5c1bf30fb1384d83cb2325

File: pattern-3/dockerfiles/build.sh (hunk 2 above)
Comment (pubudu538) [inline pattern-3/dockerfiles/build.sh]: Fixed with https://github.com/wso2/kubernetes-apim/pull/19/commits/6627082c93cd02a92d5c1bf30fb1384d83cb2325

**Pull request merged**